### PR TITLE
Add `when` attribute for `slack/notify` command

### PR DESCRIPTION
**Pull request by cayter**
### Checklist

- [x] All new jobs, commands, executors, parameters have descriptions
- [x] Examples have been added for any significant new features
- [x] README has been updated, if necessary

### Motivation, issues

Related issue: https://github.com/CircleCI-Public/slack-orb/issues/33

### Description

This PR is to introduce `when` attribute to `slack/notify` so that users can freely decide when to send the notification based on CircleCI's job status.

### CODE REVIEW

**Comment by iynere** [comment]
this is a great idea ! @cayter 

can i suggest you change the parameter type from `string` to `enum` ?

https://circleci.com/docs/2.0/reusing-config/#enum

that way, we only give folks options that are actually available

thank you !

**Comment by rdublenych** [comment]
Any news on this?

**Comment by KyleTryon** [comment]
Hello @rdublenych Thank you for contributing. Would you mind pushing another commit in order to trigger our automated testing? I believe the branch may be out of date so we are missing the statuses at the moment.

**Comment by cayter** [comment]
@rdublenych @KyleTryon @iynere Can help with reviewing the latest change? Just got time to get back to this. Sorry and thanks.

File: src/commands/notify.yml [hunk 1]
```diff
@@ -85,6 +103,12 @@ steps:
           echo "Please input your SLACK_WEBHOOK value either in the settings for this project, or as a parameter for this orb."
           exit 1
         else
+          if [[ -n "<< parameters.when >>" && << parameters.when >> != "always" ]]; then
```
Comment (KyleTryon) [inline src/commands/notify.yml]: I believe this step will need to be modified so this step itself possesses the "when:always" parameter. If the job has failed before the notification this step will not execute, only the setting failure condition step would have the chance to run.

Comment (KyleTryon) [reply src/commands/notify.yml]: @cayter ^

**Pull request closed without merging**